### PR TITLE
[datadog/connector] Create a simplified Trace to Trace connector

### PR DESCRIPTION
**Pull request by dineshg13**
**Description:** <Describe what has changed.>

Datadog Connector is creating two instances of Trace Agent, one in the trace-to-metrics pipeline and another in the traces-to-traces pipeline. The PR separates the trace-to-trace connector, simplifying the logic, this avoid un-necessary serialization. 

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30828
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30487

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: connector/datadogconnector/connector.go [hunk 1]
```diff
@@ -46,9 +45,8 @@ type connectorImp struct {
 var _ component.Component = (*connectorImp)(nil) // testing that the connectorImp properly implements the type Component interface
 
 // function to create a new connector
-func newConnector(set component.TelemetrySettings, cfg component.Config, metricsConsumer consumer.Metrics, tracesConsumer consumer.Traces) (*connectorImp, error) {
-	set.Logger.Info("Building datadog connector")
-
+func newConnector(set component.TelemetrySettings, cfg component.Config, metricsConsumer consumer.Metrics) (*connectorImp, error) {
```
Comment (mx-psi) [inline connector/datadogconnector/connector.go:48]: If this is now the traces to metrics connector, I think we should rename this and the `connectorImp` structs to reflect that

File: internal/datadog/agent.go [hunk 2]
```diff
@@ -148,9 +144,6 @@ func (p *TraceAgent) Ingest(ctx context.Context, traces ptrace.Traces) {
 		// ...the call transforms the OTLP Spans into a Datadog payload and sends the result
 		// down the p.pchan channel
 
-		// Stats will be computed for p. Mark the original resource spans to ensure that they don't
-		// get computed twice in case these spans pass through here again.
-		rspans.Resource().Attributes().PutBool(keyStatsComputed, true)
```
Comment (songy23) [inline internal/datadog/agent.go:153]: This will break datadogprocessor if anyone is still using it.

Comment (dineshg13) [reply internal/datadog/agent.go:153]: We should delete `datadogprocessor` soon. No one should be using that.

Comment (mx-psi) [reply internal/datadog/agent.go:153]: Let's merge #31037 first then :)

File: connector/datadogconnector/connector.go [hunk 3]
```diff
@@ -89,15 +86,15 @@ func getTraceAgentCfg(cfg TracesConfig) *traceconfig.AgentConfig {
 func (c *connectorImp) Start(_ context.Context, _ component.Host) error {
 	c.logger.Info("Starting datadogconnector")
 	c.agent.Start()
-	if c.metricsConsumer != nil {
-		go c.run()
-	}
+	go c.run()
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply connector/datadogconnector/connector.go:89]: Ah, right, makes sense

**Review by mx-psi — APPROVED** [review]

**Review by songy23 — APPROVED** [review]
LGTM

**Pull request merged**